




<launch>

  <arg name="robot_ip" />
	
  <!-- controller: Controller name (fs100 or dx100) -->
  <arg name="controller" default="fs100"/>

  <arg name="joint_command_topic" default="simple_joint_command"/>

  <arg name="get_joints_service" default="getJoints"/>

  <arg name="desired_pose_twist_topic" default="desired_posetwist"/>

  <arg name="set_stopped_service" default="set_stop"/>

  <arg name="clik_gain" default="0.5"/>

  <arg name="dls_joint_speed_saturation" default="3.0"/>

  <arg name="second_obj_gain" default="0.0"/>

  <arg name="start_stopped" default="true"/>

  <arg name="joint_target" default="[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]"/>

  <arg name="joint_weights" default="[1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]"/>

  <arg name="mask" default="[1, 1, 1, 1, 1, 1]"/>


  <include file="$(find motoman_interface)/launch/robot_interface_streaming_sia5d.launch">

    <arg name="robot_ip" value="$(arg robot_ip)" />
	
    <!-- controller: Controller name (fs100 or dx100) -->
    <arg name="controller" value="$(arg controller)"/>

    <arg name="in_joint_command_topic" value="$(arg joint_command_topic)"/>

    <arg name="get_joints_service" value="$(arg get_joints_service)"/>

  </include>

  <node name="motoman_clik"  pkg="sun_motoman_nodes" type="motoman_clik" output="screen">

      <param name="joint_command_topic" type="string" value="$(arg joint_command_topic)"/>

      <param name="service_get_joints" type="string" value="$(arg get_joints_service)"/>

      <param name="desired_pose_twist_topic" type="string" value="$(arg desired_pose_twist_topic)"/>

      <param name="set_stopped_service" type="string" value="$(arg set_stopped_service)"/>

      <param name="clik_gain" value="$(arg clik_gain)"/>

      <param name="dls_joint_speed_saturation" value="$(arg dls_joint_speed_saturation)"/>

      <param name="second_obj_gain" value="$(arg second_obj_gain)"/>

      <param name="start_stopped" value="$(arg start_stopped)"/>

      <rosparam param="joint_target" subst_value="true">$(arg joint_target)</rosparam>

      <rosparam param="joint_weights" subst_value="true">$(arg joint_weights)</rosparam>

      <rosparam param="mask" subst_value="true">$(arg mask)</rosparam>

</node>

</launch>
